### PR TITLE
Add AWS Auth Backend client.

### PR DESCRIPTION
**Pull request by paddycarver**
This allows us to configure the client that Vault uses when making
requests on behalf of the AWS Auth backend. See
https://www.vaultproject.io/api/auth/aws/index.html#configure-client
for more info.

### CODE REVIEW

File: vault/resource_aws_auth_backend_client.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	}
+
+	log.Printf("[DEBUG] Writing AWS auth backend client config to %q", "/auth/"+backend+"/config/client")
+	_, err := client.Logical().Write("auth/"+backend+"/config/client", data)
+	if err != nil {
+		return fmt.Errorf("error writing to %q: %s", "auth/"+backend+"/config/client", err)
+	}
+	log.Printf("[DEBUG] Wrote AWS auth backend client config to %q", "auth/"+backend+"/config/client")
+
+	d.SetId("auth/" + backend + "/config/client")
```
Comment (catsby) [inline vault/resource_aws_auth_backend_client.go]: Probably want `"auth/" + backend + "/config/client"` to come from a convenience method at this point, it's hard-coded in 4 spots just right here 😄

File: website/docs/r/aws_auth_backend_client.html.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+layout: "vault"
+page_title: "Vault: vault_aws_auth_backend_client resource"
+sidebar_current: "docs-vault-resource-aws-auth-backend-client"
+description: |-
+  Configures the client used by an AWS Auth Backend in Vault.
+---
+
+# vault\_aws\_auth\_backend\_client
+
+Configures the client used by an AWS Auth Backend in Vault.
```
Comment (catsby) [inline website/docs/r/aws_auth_backend_client.html.md]: Nit: is it helpful to say `the HTTP client` here? 🤔

Comment (paddycarver) [reply website/docs/r/aws_auth_backend_client.html.md]: Opinion: The Vault docs never reference it as an HTTP client, so I think it's better without. But am open to changing it if you feel strongly one way or another.

**Comment by paddycarver** [comment]
Think I responded to all the feedback. Good to merge @catsby? (I know you gave it a LGTM, but new code and a response to your question makes me want to just doublecheck.)

File: website/docs/r/aws_auth_backend_client.html.md (hunk 2 above)
Comment (apparentlymart) [reply website/docs/r/aws_auth_backend_client.html.md]: I'm not super-familiar with what's going on behind the scenes here, but I suggest generally that we try to use the same terminology used in the Vault documentation so it's easier for users to understand how these things relate. If Vault is considering HTTP to be an implementation detail here, then it seems right for us to do that too.

**Pull request merged**